VestingInstance.sol

```diff
@@ -0,0 +1,273 @@
+        _beneficiary = beneficiary;
+        _vestingAmount = vestingAmount;
+        _vestAmountPerPeriod = vestAmountPerPeriod;
+        _revocable = revocable;
+        _vestingPeriodSec = vestingPeriodSec;
+        _cliffStartTime = vestingStartTime.add(vestingCliff);
+        _vestingStartTime = vestingStartTime;
+        _refundDestination = refundDestination;
+        _revoker = revoker;
+    }
+
+    /**
+     * @notice Transfers available released tokens from the vesting back to beneficiary.
+     */
+    function withdraw() external onlyBeneficiary {
+        uint256 releasableAmount = _getReleasableAmount(block.timestamp);
+
+        require(releasableAmount > 0, "No unreleased tokens are due for withdraw");
+
+        _currentlyReleased = _currentlyReleased.add(releasableAmount);
+
+        getGoldToken().safeTransfer(_beneficiary, releasableAmount);
+
+        emit VestingWithdrawn(msg.sender, releasableAmount, block.timestamp);
+    }
+
+    /**
+     * @notice Allows only the revoker to revoke the vesting. Gold already vested
+     * remains in the contract, the rest is returned to the _refundDestination.
+     * @param revokeTime the revocation timestamp
+     * @dev revokeTime the revocation timestamp. If is less than the current block timestamp, it is set equal
+     */
+    function revoke(revokeTime) external onlyRevoker {
+        require(_revocable, "Revoking is not allowed");
+        require(!_revoked, "Vesting already revoked");
+
+        uint256 revokeTimestamp = revokeTime > block.timestamp ? revokeTime : block.timestamp;
+
+        uint256 balance = getGoldToken().balanceOf(address(this));
+        uint256 releasableAmount = _getReleasableAmount(revokeTimestamp);
+        uint256 refund = balance.sub(releasableAmount);
+
+        _revoked = true;
+        _revokeTime = revokeTimestamp;
+
+        getGoldToken().transfer(_refundDestination, refund);
+
+        emit VestingRevoked(msg.sender, _refundDestination, refund, _revokeTime);
+    }
+
+    /**
+     * @dev Calculates the amount that has already vested but hasn't been withdrawn (released) yet.
+     * @param timestamp the timestamp at which the calculate the releasable amount
+     */
+    function _getReleasableAmount(uint256 timestamp) private view returns (uint256) {
+        return _calculateFreeAmount(timestamp).sub(_currentlyReleased);
+    }
+
+    /**
+     * @dev Calculates the amount that has already vested.
+     * @param timestamp the timestamp at which the calculate the already vested amount
+     */
+    function _calculateFreeAmount(uint256 timestamp) private view returns (uint256) {
+        uint256 currentBalance = getGoldToken().balanceOf(address(this));
+        uint256 totalBalance = currentBalance.add(_currentlyReleased);
+
+        if (timestamp < _cliffStartTime) {
+            return 0;
+        } else if (timestamp >= _vestingStartTime.add( _vestingPeriods.mul(_vestingPeriodSec) ) || _revoked) {
```

Comment:
else if is hard to if. You can safely skip the else and start a new if statement due to the return above. I'd personaly write this as follows:
```
if (timestamp < _cliffStartTime) {
    return 0;
}

if (timestamp >= _vestingStartTime.add( _vestingPeriods.mul(_vestingPeriodSec) ) || _revoked) {
    return totalBalance
}

uint256 gradient = (timestamp.sub(_vestingStartTime)).div(_vestingPeriodSec);
return  ( (currentBalance.mul(gradient)).mul(vestAmountPerPeriod) ).div(totalBalance);
```

Reply:
absolutely, changed